src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      newState = { ...newState, ...action.extraData };
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const removeKey of action.keysToRemove) {
+        delete newState[removeKey];
+      }
+    }
+
+    if (action.type === 'clear') {
+      newState = {};
+    }
+    result.push({ ...newState });
```

Comment:
According to the checklist item 4, you should push a copy of the object to the 'result' array at the end of the loop cycle but outside of the switch block, after the current action is processed. This will avoid repetition and make the code cleaner.